src/cpp/transport/tcp/TCPServerWindows.cpp

```diff
@@ -30,17 +30,23 @@ TCPServer::TCPServer(uint16_t port, uint16_t discovery_port)
       buffer_{0},
       listener_thread_(),
       running_cond_(false),
-      messages_queue_{}
-{
-    (void) discovery_port;
-}
+      messages_queue_{},
+      discovery_server_(*processor_, port_, discovery_port)
+{}
 
 bool TCPServer::init(bool discovery_enabled)
 {
-    (void) discovery_enabled;
-
     bool rv = false;
 
+    /* Init discovery. */
+    if (discovery_enabled)
+    {
+        if (!discovery_server_.run())
+        {
+            return false;
```

Comment:
Are we using the one return function style in C++ too?

Reply:
Yes, I think so.